src/agoradatatools/process.py

```diff
@@ -18,57 +18,37 @@
 
 # TODO refactor to avoid so many if's - maybe some sort of mapping to callables
 def apply_custom_transformations(
-    datasets: dict, dataset_name: str, dataset_obj: dict
+    datasets: dict,
+    dataset_name: str,
+    dataset_obj: dict,
 ) -> Union[DataFrame, dict, None]:
-    if not isinstance(datasets, dict) or not isinstance(dataset_name, str):
+    function_info = dataset_obj.get("custom_transformations", "")
+    if (
+        not isinstance(datasets, dict)
+        or not isinstance(dataset_name, str)
+        or not function_info
+    ):
         return None
```

Comment:
This is awesome to see the simplification (And I do see it's still in Draft, so I'll just leave this 1 comment).

I would ensure that if `function_info = dataset_obj.get("custom_transformations", "")` doesn't return back a valid transformation we should write a warning message here to indicate that something went wrong. This code seems like it would fail silently and require a bit of debugging to understand what went wrong.

Reply:
Thank you, Bryan! I will add a warning message.